src/ringmesh/io/geomodel_builder_resqml.cpp

```diff
@@ -0,0 +1,968 @@
+                              << obj->getExtraMetadataStringValueAtIndex( i )
+                              << std::endl;
+                }
+            }
+            else
+            {
+                std::cout << prefix << "IS PARTIAL!" << std::endl;
+            }
+            std::cout << prefix
+                      << "--------------------------------------------------"
+                      << std::endl;
+        }
+
+    } // anonymous namespace
+
+    /****************************************************************************/
+    struct UnitInfo
+    {
+        UnitInfo()
+            : relation_top_( RELATION::CONFORMABLE ),
+              relation_base_( RELATION::CONFORMABLE )
+        {
+        }
+
+        gmge_id interface_top_;
+        gmge_id interface_base_;
+        gmge_id layer_;
+        RELATION relation_top_;
+        RELATION relation_base_;
+        std::string name_;
+    };
+
+    class GeoModelBuilderRESQMLImpl
+    {
+    public:
+        GeoModelBuilderRESQMLImpl(
+            GeoModelBuilderRESQML& builder, GeoModel3D& geomodel );
+        ~GeoModelBuilderRESQMLImpl() = default;
+
+        bool load_file();
+
+        void deserialize( EpcDocument& pck );
+        void read_strati_column( EpcDocument& pck );
+        bool read_surfaces( const EpcDocument& pck );
+        bool read_volumes( const EpcDocument& pck );
+
+    protected:
+        gmge_id find_layer( const Region3D& region ) const;
+
+    private:
+        GeoModelBuilderRESQML& builder_;
+        GeoModel3D& geomodel_;
+
+        std::map< AbstractFeatureInterpretation*, gmge_id >
+            interp_2_geo_entity_;
+        std::map< AbstractFeatureInterpretation*, UnitInfo > unit_2_info_;
+    };
+
+    GeoModelBuilderRESQMLImpl::GeoModelBuilderRESQMLImpl(
+        GeoModelBuilderRESQML& builder, GeoModel3D& geomodel )
+        : builder_( builder ), geomodel_( geomodel )
+    {
+    }
+
+    void GeoModelBuilderRESQMLImpl::deserialize( EpcDocument& pck )
+    {
+        std::string resqmlResult = pck.deserialize();
+        if( !resqmlResult.empty() )
+        {
+            std::cerr << resqmlResult << std::endl;
+            std::cout << "Press enter to continue..." << std::endl;
+            std::cin.get();
+        }
+
+        std::cout << "EpcDocument name " << pck.getName() << " in "
+                  << ( pck.getStorageDirectory().empty()
+                             ? "working directory."
+                             : pck.getStorageDirectory() )
+                  << std::endl;
+
+        unsigned int hdfProxyCount = pck.getHdfProxyCount();
+        std::cout << "There are " << pck.getHdfProxyCount()
+                  << " hdf files associated to this epc document." << std::endl;
+        for( unsigned int hdfProxyIndex = 0; hdfProxyIndex < hdfProxyCount;
+             ++hdfProxyIndex )
+        {
+            std::cout << "Hdf file relative path : "
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this usage of "::std::cout" by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS106)